### ISSUE TITLE
Pass unhandled pause menu submissions to the server

### ISSUE THREAD

**Issue by rollerozxa** [luanti-org/luanti#13928]
### Problem

It is possible to customise the pause menu using formspec prepend, but the extensibility is limited as you cannot make buttons that can trigger custom events on the server side, as the handling is all done on the client.

### Solutions

If a formspec submission is done on the pause menu and it doesn't contain any of the field checks in the C++ code, it should take the fields and submit it to the server. So if you have a button with ID `btn_custom`, you can put the following server code to make it trigger something.

```lua
minetest.register_on_player_receive_fields(function(player, formname, fields)
	if formname == "MT_PAUSE_MENU" then
		if fields.btn_custom then
			open_custom_dialog()
		end
	end
end)
```

### Alternatives

Something about (SS)CSMs and mod channels.

**Comment by red-001** [comment]
Uhm this seems kinda like a security flaw to have this as a feature, I don't think this should be expanded further, I suppose that was already brought up and ignored when it came to adding formspecs the client can't close, don't really love how this would made the change password menu unsafe by design.

**Comment by SmallJoker** [comment]
Indeed. The engine-provided formspec should be kept untouched (perhaps minor styling aside). Games should implement their own settings formspecs where necessary.
